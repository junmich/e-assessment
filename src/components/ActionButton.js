import React from 'react';

const ActionButton = ({ label, onClick }) => {
    return (
        <div className="action-button">
            <button className="default-button" onClick={onClick}>{label}</button>
        </div>
    )
};

ActionButton.defaultProps = {
    label: 'Create New'
};

export default ActionButton;