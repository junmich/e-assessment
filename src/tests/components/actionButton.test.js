import React from 'react';
import { shallow } from 'enzyme';
import ActionButton from '../../components/ActionButton';

describe('Action Button Test', () => {
    test('should render header correctly', () => {
        const wrapper = shallow(<ActionButton />);
        expect(wrapper.find('.action-button').length).toBe(1);
        expect(wrapper).toMatchSnapshot();
    });
    test('should render header with title', () => {
        const label = "My App";
        const wrapper = shallow(<ActionButton label={label}/>);
        expect(wrapper.find('.action-button').text()).toBe(label);
        expect(wrapper).toMatchSnapshot();
    });
});

