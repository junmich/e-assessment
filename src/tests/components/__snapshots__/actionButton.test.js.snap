// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Button Test should render header correctly 1`] = `
<div
  className="action-button"
>
  <button
    className="default-button"
  >
    Create New
  </button>
</div>
`;

exports[`Action Button Test should render header with title 1`] = `
<div
  className="action-button"
>
  <button
    className="default-button"
  >
    My App
  </button>
</div>
`;
